File: Linear Data Structures/Arrays/Practice Problems Using Arrays/Python/Second Largest Elements.py
def findSecondLargest(arr):
    largest = arr[0]
    secondLargest = -1
    for num in arr[1:]:
        if num > largest:
            secondLargest = largest
            largest = num
        elif num > secondLargest and num != largest:
            secondLargest = num
    return secondLargest

n = int(input("Enter the number of elements: "))
arr = [int(input(f"Enter element {i + 1}: ")) for i in range(n)]

secondLargest = findSecondLargest(arr)
print("Second largest element:", secondLargest)

File: Linear Data Structures/Arrays/Practice Problems Using Arrays/C++/Second Largest Elements.cpp
#include <iostream>
using namespace std;

int findSecondLargest(int arr[], int n) {
    int largest = arr[0], secondLargest = -1;
    for (int i = 1; i < n; i++) {
        if (arr[i] > largest) {
            secondLargest = largest;
            largest = arr[i];
        } else if (arr[i] > secondLargest && arr[i] != largest) {
            secondLargest = arr[i];
        }
    }
    return secondLargest;
}

int main() {
    int n;
    cout << "Enter the number of elements: ";
    cin >> n;
    
    int arr[n];
    cout << "Enter the elements: ";
    for (int i = 0; i < n; i++) {
        cin >> arr[i];
    }

    int secondLargest = findSecondLargest(arr, n);
    cout << "Second largest element: " << secondLargest << endl;
    return 0;
}
